### ISSUE TITLE
When will ML.NET support deeplearning network ,such as CNTK ,tensorflow

### ISSUE THREAD

**Issue by sharpwood** [dotnet/machinelearning#480]

**Comment by GalOshri** [comment]
@sharpwood, is there a particular scenario that you are targeting? Are you looking to train a deep neural network or score/evaluate an existing model? 

Related to #159 and #67.

**Comment by sharpwood** [comment]
@GalOshri  I want to train a deep neural network. Because CNTK has no high level api for C Sharp . I want to use CNTK more simply.

**Comment by hellothere33** [comment]
Same here. I have a 6 layered network composed of dense and dropout layers. I'm using Keras currently, but I would love to use C# from end to end!

**Comment by Ivanidzo4ka** [comment]
DRI RESPONSE: We do support TensorFlow/ONNX starting from 0.6 version.

**Comment by mdabros** [comment]
@CESARDELATORRE Current ML.Net deep learning efforts seems to be mostly targeting model scoring/predictions, and transfer-learning scenarios.

From the roadmap I can see the following bullets under long term development:

 - Deep Learning Support
   - Integrate with leading DNN package(s)
   - Support for transfer learning
   - Hybrid training of pipelines containing both DNN and non-DNN predictors

Does "Integrate with leading DNN package(s)" include high-level API's for training deep learning models? That is, API's for specifying network architectures/layers and learners/trainers? Or is this currently not on the roadmap?

best regards
Mads

**Comment by CESARDELATORRE** [comment]
@mdabros  - You are right, in the short-term, the current roadmap is to support multiple Deep Learning frameworks but initially only for scoring a pre-trained model.
For now, we support TensorFlow and ONNX formats.

 In the longer term, we might implement training from ML.NET API, like training a TensorFlow model, but that is not confirmed in the roadmap for the upcoming releases and will take more time, if confirmed.

**Comment by mdabros** [comment]
@CESARDELATORRE Thanks for the information, and thanks for the efforts. I hope we will one day see a deep learning training API in ML.Net :-)

**Comment by david-uk-hash** [comment]
Any update on this ....

**Comment by harishsk** [comment]
Sorry, we don't have any further update on this. We would love to learn more about your specific scenario and requirements so that we can include them in our planning.
